### PR TITLE
[DO NOT MERGE] Dummy PR to test coqbot ci minimize facilities

### PR DESCRIPTION
**Pull request by JasonGross**
Like https://github.com/coq/coq/pull/14579, created to hold the state of #14733 for debugging issues with the minimizer failing to completely minimize HoTT

### CODE REVIEW

**Comment by JasonGross** [comment]
@coqbot ci minimize ci-hott

**Comment by JasonGross** [comment]
@coqbot ci minimize ci-hott

**Comment by Zimmi48** [comment]
@coqbot ci minimize ci-hott

**Comment by Zimmi48** [comment]
@coqbot ci minimize ci-hott

**Comment by Zimmi48** [comment]
@JasonGross I've stopped the minimization of category-theory, because it looks like we're in an infinite loop (where each new run produces the same file again and again).

**Comment by JasonGross** [comment]
@coqbot resume ci minimize ci-category_theory
```
(* -*- mode: coq; coq-prog-args: ("-emacs" "-q" "-w" "-deprecated-native-compiler-option" "-R" "/github/workspace/builds/coq/coq-failing/_build_ci/category_theory" "Category" "-Q" "/github/workspace/cwd" "Top" "-Q" "/github/workspace/builds/coq/coq-failing/_install_ci/lib/coq/user-contrib/Equations" "Equations" "-Q" "/github/workspace/builds/coq/coq-failing/_install_ci/lib/coq/user-contrib/Ltac2" "Ltac2" "-top" "bug_01" "-native-compiler" "no") -*- *)
(* File reduced by coq-bug-finder from original input, then from 213 lines to 48 lines, then from 90 lines to 173 lines, then from 178 lines to 75 lines, then from 116 lines to 279 lines, then from 284 lines to 84 lines, then from 125 lines to 337 lines, then from 342 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines *)
(* coqc version 8.15+alpha compiled with OCaml 4.05.0
   coqtop version runner-0277ea0f-project-6138686-concurrent-0:/builds/coq/coq/_build/default,(HEAD detached at 0d1a2eb) (0d1a2eb7ba6a7719f37495807fb9fa49623ac075)
   Expected coqc runtime on this file: 21.290 sec *)
Axiom proof_admitted : False.
Tactic Notation "admit" := abstract case proof_admitted.
Require Category.Structure.Cartesian.Closed.
Require Category.Theory.Natural.Transformation.
Export Category.Theory.Natural.Transformation.

Section Fun.

Context {C : Category}.
Context {D : Category}.

Global Program Definition Fun : Category := {|
  obj     := C ⟶ D;
  hom     := @Transform C D;
  id      := @nat_id C D;
  compose := @nat_compose C D;

  compose_respects := @nat_compose_respects C D
|}.

End Fun.

Notation "[ C , D ]" := (@Fun C D)
  (at level 90, right associativity, format "[ C ,  D ]") : category_scope.
Set Universe Polymorphism.

Program Instance Cat : Category := {
  obj     := Category;
  hom     := @Functor;
  homset  := @Functor_Setoid;
  id      := @Id;
  compose := @Compose;

  compose_respects := @Compose_respects
}.
Export Category.Structure.Cartesian.
Export Category.Construction.Product.

Program Instance Cat_Cartesian : @Cartesian Cat := {
  product_obj := @Product;
  fork := fun _ _ _ F G =>
            {| fobj := fun x => (F x, G x)
             ; fmap := fun _ _ f => (fmap[F] f, fmap[G] f) |};
  exl := fun _ _ =>
            {| fobj := fst
             ; fmap := fun _ _ => fst |};
  exr := fun _ _ =>
            {| fobj := snd
             ; fmap := fun _ _ => snd |};
}.
Admit Obligations.
Export Category.Structure.Cartesian.Closed.

Program Instance Cat_Closed : @Closed Cat Cat_Cartesian := {
  exponent_obj := @Fun;
  exp_iso := fun A B C =>
    {| to :=
       {| morphism := fun F : A × B ⟶ C =>
          {| fobj := fun x : A =>
             {| fobj := fun y : B => F (x, y)
              ; fmap := fun J K (f : J ~{B}~> K) =>
                  fmap[F] (@id A x, f) |}
           ; fmap := fun J K (f : J ~{A}~> K) =>
             {| transform := fun L : B =>
                  fmap[F] (f, @id B L) |} |} |}
     ; from :=
       {| morphism := fun F : A ⟶ [B, C] =>
          {| fobj := fun x : A × B => F (fst x) (snd x)
           ; fmap := fun J K (f : J ~{A × B}~> K) =>
               fmap (snd f) ∘ transform[fmap[F] (fst f)] _ |} |} |}
}.
Admit Obligations.
```

**Comment by JasonGross** [comment]
(resumed minimization on a slightly smaller file to see if it can make progress when I strip out some of the code)

**Comment by JasonGross** [comment]
@coqbot minimize
```bash
git clone https://github.com/jwiegley/category-theory
export COQEXTRAFLAGS='-native-compiler no'
( cd "category-theory"
  make
  make install
)
cat > bug.v <<'_EOF'
(* -*- mode: coq; coq-prog-args: ("-emacs" "-q" "-w" "-deprecated-native-compiler-option" "-R" "/github/workspace/builds/coq/coq-failing/_build_ci/category_theory" "Category" "-Q" "/github/workspace/cwd" "Top" "-Q" "/github/workspace/builds/coq/coq-failing/_install_ci/lib/coq/user-contrib/Equations" "Equations" "-Q" "/github/workspace/builds/coq/coq-failing/_install_ci/lib/coq/user-contrib/Ltac2" "Ltac2" "-top" "bug_01" "-native-compiler" "no") -*- *)
(* File reduced by coq-bug-finder from original input, then from 213 lines to 48 lines, then from 90 lines to 173 lines, then from 178 lines to 75 lines, then from 116 lines to 279 lines, then from 284 lines to 84 lines, then from 125 lines to 337 lines, then from 342 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines, then from 125 lines to 85 lines *)
(* coqc version 8.15+alpha compiled with OCaml 4.05.0
   coqtop version runner-0277ea0f-project-6138686-concurrent-0:/builds/coq/coq/_build/default,(HEAD detached at 0d1a2eb) (0d1a2eb7ba6a7719f37495807fb9fa49623ac075)
   Expected coqc runtime on this file: 21.290 sec *)
Declare ML Module "ltac_plugin".
Module Export AdmitTactic.
Module Import LocalFalse.
Inductive False : Prop := .
End LocalFalse.
Axiom proof_admitted : False.
Tactic Notation "admit" := abstract case proof_admitted.
End AdmitTactic.
Require Coq.Init.Datatypes.
Require Coq.Program.Program.
Require Category.Lib.Foundation.
Require Coq.Classes.CEquivalence.
Require Coq.Classes.CRelationClasses.
Require Coq.Classes.CMorphisms.
Require Category.Lib.Setoid.
Require Coq.Bool.Bool.
Require Category.Lib.Tactics.
Require Coq.NArith.NArith.
Require Coq.Lists.List.
Require Category.Lib.Datatypes.
Require Category.Lib.
Require Category.Theory.Category.
Require Category.Theory.Morphisms.
Require Category.Theory.Isomorphism.
Require Category.Structure.Terminal.
Require Category.Structure.Cartesian.
Require Category.Construction.Opposite.
Require Category.Structure.Initial.
Require Category.Theory.Functor.
Require Category.Theory.Functor.Endo.
Require Category.Theory.Naturality.
Require Category.Construction.Product.
Require Category.Functor.Bifunctor.
Require Category.Structure.Monoidal.
Require Category.Instance.Sets.
Require Category.Structure.Cartesian.Closed.
Require Category.Theory.Natural.Transformation.

Axiom proof_admitted : False.
Tactic Notation "admit" := abstract case proof_admitted.
Export Category.Theory.Natural.Transformation.

Section Fun.

Context {C : Category}.
Context {D : Category}.

Global Program Definition Fun : Category := {|
  obj     := C ⟶ D;
  hom     := @Transform C D;
  id      := @nat_id C D;
  compose := @nat_compose C D;

  compose_respects := @nat_compose_respects C D
|}.

End Fun.

Notation "[ C , D ]" := (@Fun C D)
  (at level 90, right associativity, format "[ C ,  D ]") : category_scope.
Set Universe Polymorphism.

Program Instance Cat : Category := {
  obj     := Category;
  hom     := @Functor;
  homset  := @Functor_Setoid;
  id      := @Id;
  compose := @Compose;

  compose_respects := @Compose_respects
}.
Export Category.Structure.Cartesian.
Export Category.Construction.Product.

Program Instance Cat_Cartesian : @Cartesian Cat := {
  product_obj := @Product;
  fork := fun _ _ _ F G =>
            {| fobj := fun x => (F x, G x)
             ; fmap := fun _ _ f => (fmap[F] f, fmap[G] f) |};
  exl := fun _ _ =>
            {| fobj := fst
             ; fmap := fun _ _ => fst |};
  exr := fun _ _ =>
            {| fobj := snd
             ; fmap := fun _ _ => snd |};
}.
Admit Obligations.
Export Category.Structure.Cartesian.Closed.

Program Instance Cat_Closed : @Closed Cat Cat_Cartesian := {
  exponent_obj := @Fun;
  exp_iso := fun A B C =>
    {| to :=
       {| morphism := fun F : A × B ⟶ C =>
          {| fobj := fun x : A =>
             {| fobj := fun y : B => F (x, y)
              ; fmap := fun J K (f : J ~{B}~> K) =>
                  fmap[F] (@id A x, f) |}
           ; fmap := fun J K (f : J ~{A}~> K) =>
             {| transform := fun L : B =>
                  fmap[F] (f, @id B L) |} |} |}
     ; from :=
       {| morphism := fun F : A ⟶ [B, C] =>
          {| fobj := fun x : A × B => F (fst x) (snd x)
           ; fmap := fun J K (f : J ~{A × B}~> K) =>
               fmap (snd f) ∘ transform[fmap[F] (fst f)] _ |} |} |}
}.
Admit Obligations.
_EOF
coqc -q bug.v
```

**Comment by JasonGross** [comment]
@coqbot ci minimize ci-mtac2 ci-mtac2

**Comment by JasonGross** [comment]
@coqbot ci minimize ci-mtac2

**Comment by Zimmi48** [comment]
FTR the ci-category-theory job had to be stopped because it was in an infinite loop...

**Comment by JasonGross** [comment]
Now tracking #15518

**Comment by JasonGross** [comment]
@coqbot ci minimize

**Comment by JasonGross** [comment]
Now tracking https://github.com/coq/coq/pull/15653#issuecomment-1034144211, need to debug ci-iris

**Comment by JasonGross** [comment]
@coqbot ci minimize ci-iris

**Comment by JasonGross** [comment]
@coqbot ci minimize  ci-iris

**Comment by JasonGross** [comment]
@coqbot resume ci minimize ci-iris
```
(* -*- mode: coq; coq-prog-args: ("-emacs" "-q" "-w" "-deprecated-hint-rewrite-without-locality" "-w" "-deprecated-typeclasses-transparency-without-locality" "-w" "-deprecated-native-compiler-option" "-Q" "/github/workspace/cwd" "Top" "-Q" "/github/workspace/builds/coq/coq-failing/_build_ci/stdpp/theories" "stdpp" "-Q" "/github/workspace/builds/coq/coq-failing/_install_ci/lib/coq/user-contrib/Autosubst" "Autosubst" "-Q" "/github/workspace/builds/coq/coq-failing/_install_ci/lib/coq/user-contrib/Ltac2" "Ltac2" "-top" "gmap" "-test-mode" "-native-compiler" "ondemand") -*- *)
(* File reduced by coq-bug-minimizer from original input, then from 85 lines to 31 lines, then from 44 lines to 440 lines, then from 445 lines to 61 lines, then from 74 lines to 440 lines, then from 445 lines to 76 lines, then from 89 lines to 273 lines, then from 278 lines to 107 lines, then from 120 lines to 532 lines, then from 537 lines to 99 lines, then from 112 lines to 3566 lines, then from 3571 lines to 125 lines, then from 138 lines to 710 lines, then from 715 lines to 125 lines, then from 138 lines to 1513 lines, then from 1513 lines to 126 lines, then from 139 lines to 555 lines, then from 560 lines to 141 lines, then from 154 lines to 5269 lines, then from 5274 lines to 143 lines, then from 149 lines to 144 lines *)
(* coqc version 8.16+alpha compiled with OCaml 4.13.0
   coqtop version runner-zxwgkjap-project-6138686-concurrent-0:/builds/coq/coq/_build/default,(HEAD detached at 9b9d920) (9b9d9203517ca574dbacbd82f0fdc4c41f4b5cd1)
   Modules that could not be inlined: stdpp.numbers
   Expected coqc runtime on this file: 0.910 sec *)
Require stdpp.numbers.

Axiom proof_admitted : False.
Tactic Notation "admit" := abstract case proof_admitted.
Export stdpp.numbers.
Global Instance list_fmap : FMap list.
Admitted.
Local Open Scope positive.

Class Countable A `{EqDecision A} := {
  encode : A → positive;
  decode : positive → option A;
  decode_encode x : decode (encode x) = Some x
}.
Global Program Instance N_countable : Countable N := {|
  encode x := match x with N0 => 1 | Npos p => Pos.succ p end;
  decode p := if decide (p = 1) then Some 0%N else Some (Npos (Pos.pred p))
|}.
Admit Obligations.
Global Program Instance nat_countable : Countable nat :=
  {| encode x := encode (N.of_nat x); decode p := N.to_nat <$> decode p |}.
Admit Obligations.

Local Close Scope positive.

Class FinMapToList K A M := map_to_list: M → list (K * A).
Definition diag_None {A B C} (f : option A → option B → option C)
    (mx : option A) (my : option B) : option C.
Admitted.

Class FinMap K M `{FMap M, ∀ A, Lookup K A (M A), ∀ A, Empty (M A), ∀ A,
    PartialAlter K A (M A), OMap M, Merge M, ∀ A, FinMapToList K A (M A),
    EqDecision K} := {
  map_eq {A} (m1 m2 : M A) : (∀ i, m1 !! i = m2 !! i) → m1 = m2;
  lookup_empty {A} i : (∅ : M A) !! i = None;
  lookup_partial_alter {A} f (m : M A) i :
    partial_alter f i m !! i = f (m !! i);
  lookup_partial_alter_ne {A} f (m : M A) i j :
    i ≠ j → partial_alter f i m !! j = m !! j;
  lookup_fmap {A B} (f : A → B) (m : M A) i : (f <$> m) !! i = f <$> m !! i;
  NoDup_map_to_list {A} (m : M A) : NoDup (map_to_list m);
  elem_of_map_to_list {A} (m : M A) i x :
    (i,x) ∈ map_to_list m ↔ m !! i = Some x;
  lookup_omap {A B} (f : A → option B) (m : M A) i :
    omap f m !! i = m !! i ≫= f;
  lookup_merge {A B C} (f : option A → option B → option C) (m1 : M A) (m2 : M B) i :
    merge f m1 m2 !! i = diag_None f (m1 !! i) (m2 !! i)
}.
Global Instance map_insert `{PartialAlter K A M} : Insert K A M.
Admitted.

Record mapset (M : Type → Type) : Type :=
  Mapset { mapset_car: M (unit : Type) }.

Section mapset.
Context `{FinMap K M}.
Global Instance mapset_elem_of: ElemOf K (mapset M).
admit.
Defined.
Global Instance mapset_union: Union (mapset M).
admit.
Defined.
Global Instance mapset_intersection: Intersection (mapset M).
admit.
Defined.
Global Instance mapset_difference: Difference (mapset M).
admit.
Defined.
Global Instance mapset_elements: Elements K (mapset M).
exact (λ X,
  let (m) := X in (map_to_list m).*1).
Defined.

End mapset.

Inductive Pmap_raw (A : Type) : Type :=
  | PLeaf: Pmap_raw A
  | PNode: option A → Pmap_raw A → Pmap_raw A → Pmap_raw A.
Global Arguments PLeaf {_} : assert.
Global Arguments PNode {_} _ _ _ : assert.

Fixpoint Pmap_wf {A} (t : Pmap_raw A) : bool :=
  match t with
  | PLeaf => true
  | PNode None PLeaf PLeaf => false
  | PNode _ l r => Pmap_wf l && Pmap_wf r
  end.

Inductive Pmap (A : Type) : Type :=
  PMap { pmap_car : Pmap_raw A; pmap_prf : Pmap_wf pmap_car }.
Definition gmap_wf K `{Countable K} {A} (m : Pmap A) : Prop.
Admitted.
Record gmap K `{Countable K} A := GMap {
  gmap_car : Pmap A;
  gmap_prf : gmap_wf K gmap_car
}.
Global Instance gmap_empty `{Countable K} {A} : Empty (gmap K A).
Admitted.
Global Instance gmap_partial_alter `{Countable K} {A} :
    PartialAlter K A (gmap K A).
Admitted.
Global Instance gmap_to_list `{Countable K} {A} : FinMapToList K A (gmap K A).
Admitted.

Definition gset K `{Countable K} := mapset (gmap K).

Section gset.
  Context `{Countable K}.
Global Instance gset_empty : Empty (gset K).
Admitted.
Global Instance gset_singleton : Singleton K (gset K).
Admitted.
Global Instance gset_dom {A} : Dom (gmap K A) (gset K).
Admitted.
End gset.

Goal {[1; 2; 3]} =@{gset nat} ∅.
Proof.
  Fail progress simpl.
  Fail progress cbn.
  Show.
Abort.

Goal elements (C := gset nat) {[1; 2; 3]} = [].
Proof.
  Fail progress simpl.
  Fail progress cbn.
  Show.
Abort.

Goal
  {[1; 2; 3]} ∖ {[ 1 ]} ∪ {[ 4 ]} ∩ {[ 10 ]} =@{gset nat} ∅ ∖ {[ 2 ]}.
Proof.
  Fail progress simpl.
  Fail progress cbn.
  Show.
Abort.

Goal 1 ∈ dom (M := gmap nat nat) (gset nat) (<[ 1 := 2 ]> ∅).
Proof.
  Fail progress simpl.
  Fail progress cbn.
```

**Comment by Zimmi48** [comment]
Only for testing coq/bot#200: @coqbot merge now

**Comment by JasonGross** [comment]
@Zimmi48 Is there no draft PR protection?

**Comment by Zimmi48** [comment]
Indeed, there is no draft protection implemented yet (this is the most important unchecked point of https://github.com/coq/bot/issues/95) but it might be the case that it still isn't possible to merge a draft PR because of GitHub's own draft protection (since internally, coqbot relies on the "merge button" API).

Note that even if such draft protection was implemented, it wouldn't show up as a reason for not merging here because this bullet-point list is based only on the info that we can get from the `issue_comment` webhook (and it doesn't include any PR-specific data such as reviews or draft status). Coqbot only queries for the missing data once these first checks have passed. This is why it also doesn't mention "It hasn't been approved yet." in the list of reasons. We could change the code to do this query in all cases and report on a more complete list of reasons.

**Comment by Zimmi48** [comment]
@coqbot ci minimize ci-iris (checking that we have not broken it again)

**Comment by Zimmi48** [comment]
@coqbot ci minimize

**Comment by Alizter** [comment]
@coqbot ci minimize

**Comment by Zimmi48** [comment]
There seems to be an issue, but it's not on coqbot-side. It's just GitHub is not delivering any webhooks for these comments.

**Comment by Zimmi48** [comment]
It seems like the comments were finally all delivered.

**Comment by Alizter** [comment]
And for future reference, GitHub had its webhooks disrupted today for whatever reason.

**Comment by Alizter** [comment]
@coqbot merge

**Comment by Alizter** [comment]
@cqobot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot merge

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

**Comment by Alizter** [comment]
@coqbot bench

File: dev/bench/bench.sh [hunk 1]
```diff
@@ -55,7 +55,7 @@ echo $PWD
 : "${old_coq_opam_archive_git_branch:=master}"
 : "${num_of_iterations:=1}"
 : "${timeout:=3h}"
-: "${coq_opam_packages:=coq-bignums coq-hott coq-performance-tests-lite coq-engine-bench-lite coq-mathcomp-ssreflect coq-mathcomp-fingroup coq-mathcomp-algebra coq-mathcomp-solvable coq-mathcomp-field coq-mathcomp-character coq-mathcomp-odd-order coq-math-classes coq-corn coq-flocq3 coq-compcert coq-metacoq-template coq-metacoq-pcuic coq-metacoq-safechecker coq-metacoq-erasure coq-metacoq-translations coq-geocoq coq-color coq-coqprime coq-coqutil coq-bedrock2 coq-rewriter coq-fiat-core coq-fiat-parsers coq-fiat-crypto coq-unimath coq-coquelicot coq-iris-examples coq-verdi coq-verdi-raft coq-fourcolor coq-rewriter-perf-SuperFast coq-perennial coq-vst coq-category-theory}"
+: "${coq_opam_packages:=coq-mathcomp-fingroup}"
```
Comment (SkySkimmer) [inline dev/bench/bench.sh:58]: Why fingroup?

Comment (Alizter) [reply dev/bench/bench.sh:58]: It took the least amount of time

Comment (Alizter) [reply dev/bench/bench.sh:58]: And I needed one job to work since I am checking the line timing data.

Comment (SkySkimmer) [reply dev/bench/bench.sh:58]: but it depends on ssreflect

Comment (Alizter) [reply dev/bench/bench.sh:58]: That's a good point. But it really doesn't matter here.

**Comment by Alizter** [comment]
@coqbot bench

**Comment by ana-borges** [comment]
@coqbot bench

**Pull request closed without merging**